packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -124,22 +153,175 @@ const AccountRecoveryResetPassword = ({
-    </>
+
+      {linkStatus === 'damaged' && (
+        <LinkDamaged {...{ linkType: 'reset-password' }} />
+      )}
+
+      {linkStatus === 'expired' && (
+        <LinkExpired {...{ linkType: 'reset-password', resendLinkHandler }} />
+      )}
+    </AppLayout>
   );
+
+  /**
+   * Determines starting state for component
+   */
+  function getInitialState() {
+    let email = '';
+    let linkStatus: LinkStatus = 'valid';
+    let forceAuth = false;
+    let supportsRecovery = true;
+    let contextError: ContextValidationErrors | null = null;
+
+    try {
+      email = verificationInfo.email || '';
+
+      forceAuth = !!verificationInfo.forceAuth;
+
+      if (!verificationInfo.isValid()) {
+        supportsRecovery = false;
+        linkStatus = 'damaged';
+      } else if (!accountRecoveryKeyInfo.isValid()) {
+        supportsRecovery = false;
+      } else if (accountRecoveryKeyInfo.lostRecoveryKey === true) {
+        supportsRecovery = false;
+      }
+    } catch (err) {
+      if (err instanceof ContextValidationErrors) {
+        contextError = err;
+        linkStatus = 'damaged';
+      }
+    }
+
+    return {
+      email,
+      linkStatus,
+      forceAuth,
+      supportsRecovery,
+      contextError,
+    };
+  }
+
+  async function onSubmit(data: FormData) {
+    const password = data.newPassword;
+
+    try {
+      await account.resetPasswordWithRecoveryKey(
+        {
+          password,
+          ...verificationInfo,
+          ...accountRecoveryKeyInfo
+        }
+      );
+
+      // FOLLOW-UP: I don't see functionality in settings
+      await account.setLastLogin(Date.now());
+
+      // FOLLOW-UP: It seems like the account class will now refresh itself, so I don't think
+      //            these are necessary anymore?
+      // storageContext.set('currentAccountUid', account.uid);
+      // await account.refresh('account');
+
+      // FOLLOW-UP: No equivalent yet in settings
+      notifier.onAccountSignIn(account);
+
+      relier.resetPasswordConfirm = true;
+      logViewEvent(viewName, 'verification.success');
+
+      // FOLLOW-UP: Broker not yet implemented
+      await broker.invokeBrokerMethod('afterCompleteResetPassword', account);
+      alertSuccess();
+      navigateAway();
+    } catch (err) {
+      if (AuthUiErrors['INVALID_TOKEN'].errno === err.errno) {
+        // TODO: Is this needed? Had to add the a method to support this.
+        // BEFORE: this.logError(err);
+        logErrorEvent({ viewName, ...err });
+        setLinkStatus('expired');
+      } else {
+        // Context validation errors indicate a bad state in either the url query or
+        // maybe storage. In these cases show an alert bar and let the error keep bubbling
+        // up.
+        if (err instanceof ContextValidationErrors) {
+          alertInvalidContext(err);
+        } else {
+          logErrorEvent(err);
+          const msg = ftlMsgResolver.getMsg(
+            'unexpected-error-encountered',
```

Comment:
Is this in an .ftl file somewhere?
I think it's the only instance of "unexpected-error-encountered" that I saw in this PR (and the repo):
```sh
git grep -n "unexpected-error-encountered" | cat

packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx:268:            'unexpected-error-encountered',
```

Reply:
Added to ftl file and id slightly altered.